### ISSUE TITLE
TinyMCE not initialized on repeater of repeater

### ISSUE THREAD

**Issue by cmachu** [siteorigin/so-widgets-bundle#745]
When I use repeater of repeater with TinyMCE editor inside, new elements of repeater are without rich edit functions. They are appear as normal textarea. After save, TinyMCE editor is normal initialized. JS console is empty on newest versions.

Wordress 4.9.8
Site Origin Widgets Bundle 1.12.1
Page Builder by SiteOrigin 2.8.2

On previous versions of Site Origin Widgets Bundle, JS console return error: `tinymce.min.js?ver=4603-20170530:14 Uncaught TypeError: Cannot read property 'onpageload' of undefined`

Form elements:
```        
$form_options = array(
            'columns' => array(
                'type' => 'repeater',
                'label' => __('Rows', 'quality-systems'),
                'item_name' => __('Row cells', 'siteorigin-widgets'),
                'item_label' => array(
                    'selector' => "[id*='repeat_text']",
                    'update_event' => 'change',
                    'value_method' => 'val'
                ),
                'fields' => array(
                    'cells' => array(
                        'type' => 'repeater',
                        'label' => __('Cells', 'quality-systems'),
                        'item_name' => __('Single cell', 'siteorigin-widgets'),
                        'item_label' => array(
                            'selector' => "[id*='repeat_text']",
                            'update_event' => 'change',
                            'value_method' => 'val'
                        ),
                        'fields' => array(
                            'cell_value' => array(
                                'type' => 'tinymce',
                                'label' => __('Cell content', 'quality-systems'),
                                'rows' => 10,
                            )
                        )
                    )
                )
            )
        );
```
Valid form, initialized after save:

![screenshot from 2018-09-10 12-31-01](https://user-images.githubusercontent.com/6084116/45292660-26295e80-b4f6-11e8-912c-5045466114fb.png)

New form, without rich editing, when click "ADD" on repeater:

![screenshot from 2018-09-10 12-31-28](https://user-images.githubusercontent.com/6084116/45292697-3d684c00-b4f6-11e8-8b77-423ad5983f08.png)

**Comment by Misplon** [comment]
Hi @cmachu Thanks for posting. There are a few open tasks being worked on today and tomorrow. As soon as those are completed I've asked @AlexGStapleton to take a look for us and get back to you. Thanks again.

**Comment by AlexGStapleton** [comment]
I can confirm this issue. [Reference video](https://drive.google.com/a/siteorigin.com/uc?id=1yzKFc8fvvFd1BwblwaaWB7az8IDbpH06).

**Comment by AlexGStapleton** [comment]
I've run a few additional tests and this issue appears to be as a result of [the repeater id](https://github.com/siteorigin/so-widgets-bundle/blob/1.12.1/base/js/admin.js#L651) not updating. I can't seem to spot why - I suspect I'll likely need to come back to this later to spot it.

**Comment by aswiercz** [comment]
Hi Alex,

I manage the project in which the very addon has been used. I therefore was wondering if you guys have already found the root and completed the fix? Paweł and I can't wait:)

Thanks for update!

Best,
Agnieszka

**Comment by AlexGStapleton** [comment]
Hi Agnieszka,

Unfortunately, I haven't been able to find the source of this issue yet. Really sorry about this.

**Comment by AlexGStapleton** [comment]
@braamgenis any ideas?

**Comment by stentibbing** [comment]
Hi

I've got the same problem. I've got nested repeaters and new elements inside repeater-repeater are missing tinymce. Only after saving it appears. Based on console output I assume textareas are not getting the IDs assigned:

[DOM] Found 5 elements with non-unique id #widget-sunny-faq-widget-c46-topics-questions-question-_id_: (More info: https://goo.gl/9p2vKq) 

Any advice ?

**Comment by stentibbing** [comment]
I've noticed that tinymce is applied only for the new elements in the forst repeater-repeater textbox. 2nd, 3rd etc. repeater-repeater boxes are missing tinymce.

Also not sure if releated but these dom elements with missing ID are nested inside invisible elements (siteorigin-widget-field-repeater-item-html) following the add div (siteorigin-widget-field-repeater-add).

**Comment by AlexGStapleton** [comment]
Hi @stentibbing,

I've run a few tests and unfortunately, I'm unable to replicate this issue. Can you please provide me with a copy of your widget code? This will allow me to run tests using a widget that is affected by this issue and (hopefully) be able to replicate this issue. I have no preference for how you send me your widget's code so whatever works for you is fine. If you're after a recommendation, try wetransfer.com (please note that no email is required).

Kind regards,
Alex

**Comment by stentibbing** [comment]
[sunny-faq-widget.zip](https://github.com/siteorigin/so-widgets-bundle/files/5014218/sunny-faq-widget.zip)

Hi Alex

Thanks for the attention. I've attached the widget.

**Comment by AlexGStapleton** [comment]
Hi,

I just tried the provided widget and it's working without issue for me - no error, and TinyMCE is working as expected.
Can you please try disabling all plugins except for our plugins and if it's being added by a plugin, that plugin. Please also try switching to a default theme.

![](https://i.imgur.com/bKQrnYo.png)

Kind regards,
Alex

**Comment by stentibbing** [comment]
Hi Alex

I tried disabling everything aside from siteorigin page builder and widget bundle, and switched to default theme. I also copied the widget under widget bundle to test it with default theme but nothing, still the same issue. For me the issue is not there in the first repeater (Topic). It occurs if I enter a topic, fill it with one question (repeater-repeater) and then create a new topic and try to fill it with a new question.

![image](https://user-images.githubusercontent.com/24324812/89345413-097d0d80-d6b0-11ea-8973-ef085ee53140.png)

**Comment by AlexGStapleton** [comment]
The widget in your screenshot contains a lot of data. It's possible that something added using the widget may be causing this issue. Are you able to replicate this issue with a new instance of the widget?

Can you please send me an export of the layout that contains that page? You can export a page by opening it in the editor and then clicking the Layouts button in the Page Builder toolbar. Click Import/Export and export the layout. Please upload the layout to [GitHub Gist](https://gist.github.com/).

**Comment by stentibbing** [comment]
Hi

I'm sorry for the long delay. Here is the gist:
https://gist.github.com/stentibbing/8a9dec89b2957b9a658f0e6345a3b955

I'll try a new Wordpress install and the widget asap.

**Comment by AlexGStapleton** [comment]
Thanks. Can you please provide me with a copy of the widget you're using?

**Comment by stentibbing** [comment]
[sunny-woo-widget.zip](https://github.com/siteorigin/so-widgets-bundle/files/5189898/sunny-woo-widget.zip)

**Comment by stentibbing** [comment]
[sunny-faq-widget.zip](https://github.com/siteorigin/so-widgets-bundle/files/5189903/sunny-faq-widget.zip)

**Comment by stentibbing** [comment]
Hi Alex

I was able to replicate this with vanilla Wordpress in docker container with just two plugins (Siteorigin PageBuilder and Siteorigin Widgets Bundle installed).

To replicate I created a topic, entered 3 questions under the topic, created a new topic and for the question in the second topic tinymce was not present. I did not save the page at any point. Please see the attached screenshot.

![Capture](https://user-images.githubusercontent.com/24324812/92627284-ced04b80-f2d3-11ea-85dc-9946f8443d84.PNG)

**Comment by Misplon** [comment]
[so-widgets-bundle.1.7.8-beta.zip](https://github.com/siteorigin/so-widgets-bundle/files/5202736/so-widgets-bundle.1.7.8-beta.zip)

Hi, please, find a build from the PR above. I haven't tested it yet.

**Comment by Misplon** [comment]
That should have been 1.17.8-beta as the version number. Here is the beta with the correct version number.

[so-widgets-bundle.1.17.8-beta.zip](https://github.com/siteorigin/so-widgets-bundle/files/5203991/so-widgets-bundle.1.17.8-beta.zip)

**Comment by stentibbing** [comment]
Hi,

Could it be this version affects state handlers somehow ? Another widget of mine is not handling 'select' state handler correctly...

**Comment by AlexGStapleton** [comment]
It's unlikely. This change didn't alter how states work and will only affect the TinyMCE field. It's possible the issue you're facing is linked to https://github.com/siteorigin/so-widgets-bundle/pull/1117 though

Please open a new issue with more information about the issue you're facing, and include a copy of your widget.

**Comment by stentibbing** [comment]
Could you test if this works for the beta for you ? I get random showing all or nothing...

https://pastebin.com/JJNRGv96

**Comment by AlexGStapleton** [comment]
Please [create a new issue](https://github.com/siteorigin/so-widgets-bundle/issues/new) and tag me (@AlexGStapleton)  and I'll take a look shortly.
